metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/StellarStatistics.java

```diff
@@ -0,0 +1,207 @@
+   * which is then used to calculate the summary statistic.  There are some
+   * summary statistics like kurtosis and percentiles, that can only
+   * be calculated with this.
+   *
+   * This implementation is used if the windowSize > 0.
+   */
+  private DescriptiveStatistics descStats;
+
+  /**
+   * SummaryStatistics can be used for summary statistics that only
+   * require a single pass over the input data.  This has the advantage
+   * of being less memory intensive, but not all summary statistics are
+   * available.
+   *
+   * This implementation is used if the windowSize == 0.
+   */
+  private SummaryStatistics summStats;
+
+  /**
+   * @param windowSize The number of input data elements to maintain in memory.  If
+   *                   windowSize == 0, then no data elements will be maintained in
+   *                   memory.
+   */
+  public StellarStatistics(int windowSize) {
+
+    // only one of the underlying implementation classes will be used at a time
+    if(windowSize > 0) {
+      descStats = new DescriptiveStatistics(windowSize);
+    } else {
+      summStats = new SummaryStatistics();
+    }
+  }
+
+  public void addValue(double value) {
+    if(descStats != null) {
+      descStats.addValue(value);
+    } else {
+      summStats.addValue(value);
+    }
+  }
+
+  public long getCount() {
+    if(descStats != null) {
+      return descStats.getN();
+    } else {
+      return summStats.getN();
+    }
+  }
+
+  public double getMin() {
+    if(descStats != null) {
+      return descStats.getMin();
+    } else {
+      return summStats.getMin();
+    }
+  }
+
+  public double getMax() {
+    if(descStats != null) {
+      return descStats.getMax();
+    } else {
+      return summStats.getMax();
+    }
+  }
+
+  public double getMean() {
+    if(descStats != null) {
+      return descStats.getMean();
+    } else {
+      return summStats.getMean();
+    }
+  }
+
+  public double getSum() {
+    if(descStats != null) {
+      return descStats.getSum();
+    } else {
+      return summStats.getSum();
+    }
+  }
+
+  public double getVariance() {
+    if(descStats != null) {
+      return descStats.getVariance();
+    } else {
+      return summStats.getVariance();
+    }
+  }
+
+  public double getStandardDeviation() {
+    if(descStats != null) {
+      return descStats.getStandardDeviation();
+    } else {
+      return summStats.getStandardDeviation();
+    }
+  }
+
+  public double getGeometricMean() {
+    if(descStats != null) {
+      return descStats.getGeometricMean();
+    } else {
+      return summStats.getGeometricMean();
+    }
+  }
+
+  public double getPopulationVariance() {
+    if(descStats != null) {
+      return descStats.getPopulationVariance();
+    } else {
+      return summStats.getPopulationVariance();
+    }
+  }
+
+  public double getSecondMoment() {
+    if(descStats != null) {
+      throw new NotImplementedException("second moment not available if 'windowSize' > 0");
```

Comment:
@cestella pointed out that second moment is the same as variance.  The naming between DescriptiveStatistics and SummaryStatistics just differs.  Just added a commit that removes `STATS_SECOND_MOMENT` as it is redundant.  Thanks!